packages/react-router/src/routeRegistry.ts

```diff
@@ -66,22 +65,40 @@ export class RouteRegistry {
 
         // Create a new array so the routes array is immutable.
         this.#routes = [...this.#routes, ...routes];
+
+        let completedPendingRegistrations;
+
+        routes.forEach(x => {
+            // Since the layoutPath is used as the key for pending routes and the
+            // layoutPath is the same as the index key, an index key for the route must be created
+            // to retrieve the route pending registrations.
+            const key = createIndexKey(x, "/");
+
+            completedPendingRegistrations = this.#tryRegisterPendingRoutes(key);
+        });
+
+        return {
+            registrationStatus: "registered",
```

Comment:
I feel like registrationStatus should be an enum instead of magic strings throughout the code

Reply:
No, with TS it is highly recommend to be favor union type over enum type.

Enum is usually considered a mistake in TS.

And by the way, it is not a magic string because the value are validated by TS since it's a union.

Reply:
Oh is there a union I didn't see, alright then